rest/rest_collection.h

```diff
@@ -117,7 +117,7 @@ struct BackgroundTaskBase {
     /** When destroying a background task, we make sure it's cancelled or
         finished so that there is nothing outstanding.
     */
-    ~BackgroundTaskBase();
+    virtual ~BackgroundTaskBase();
```

Comment:
It will only cause a problem if the derived class has a non-trivial destructor :)  but still scary